mi/dataset/parser/metbk_ct_dcl.py

```diff
@@ -0,0 +1,277 @@
+#!/usr/bin/env python
+
+"""
+@package mi.dataset.parser.metbk_ct_dcl
+@file marine-integrations/mi/dataset/parser/metbk_ct_dcl.py
+@author Tim Fisher
+@brief Parser for the metbk_ct_dcl dataset driver
+
+This file contains code for the metbk_ct_dcl parser and code to produce data particles.
+For recovered data, there is one parser which produces one type of data particle.
+
+The input file is hex and contains 1 type of record.
+All detail records are 20 bytes separated by a newline.
+Each detail record contains temperature, conductivity and time.
+Only sensor data records produce particles if properly formed.
+Mal-formed sensor data records and all metadata records produce no particles.
+
+Release notes:
+
+Initial Release
+"""
+
+import binascii
+import re
+
+from mi.core.common import BaseEnum
+from mi.core.exceptions import \
+    DatasetParserException, \
+    RecoverableSampleException, \
+    SampleException, \
+    UnexpectedDataException
+
+from mi.core.instrument.dataset_data_particle import DataParticle
+from mi.core.log import get_logger
+log = get_logger()
+
+from mi.dataset.dataset_parser import SimpleParser
+from mi.dataset.parser.utilities import zulu_timestamp_to_ntp_time
+
+INDUCTIVE_ID_KEY = 'inductive_id'
+NEW_LINE = r'[\n\r]+'             # Handle any type of new line
+
+# For Recovered CT files, the serial number is in the Configuration XML section.
+REC_CT_SERIAL_REGEX = r'^'        # At the beginning of the record
+REC_CT_SERIAL_REGEX += r'\* <HardwareData DeviceType=\'SBE37SM-RS485\' SerialNumber=\''
+REC_CT_SERIAL_REGEX += r'(\d+)'   # Serial number is any number of digits
+REC_CT_SERIAL_REGEX += r'\'>'     # the rest of the XML syntax
+REC_CT_SERIAL_MATCHER = re.compile(REC_CT_SERIAL_REGEX)
+
+# The REC_CT_SERIAL_MATCHER produces the following group:
+REC_CT_SERIAL_GROUP_SERIAL_NUMBER = 1
+
+# The end of the Configuration XML section is denoted by a *END* record.
+REC_CT_CONFIGURATION_END = r'^'                    # At the beginning of the record
+REC_CT_CONFIGURATION_END += r'\*END\*'             # *END*
+REC_CT_CONFIGURATION_END += NEW_LINE               # separated by a new line
+REC_CT_CONFIGURATION_END_MATCHER = re.compile(REC_CT_CONFIGURATION_END)
+
+# Recovered CT Data record (hex ascii):
+REC_CT_SAMPLE_BYTES = 21                # includes record separator
+
+REC_CT_REGEX = b'([0-9a-fA-F]{6})'      # Temperature
+REC_CT_REGEX += b'([0-9a-fA-F]{6})'     # Conductivity
+REC_CT_REGEX += b'([0-9a-fA-F]{8})'     # Time in seconds since 1/1/2000
```

Comment:
Its frustrating that the manual doesn't specifically call out these numbers (6, 6, 8), but as far as I can tell what we're seeing is the data from one of the formats on pages 44-46 (with most of the extra parameters turned off so they don't appear) after it has been transformed into a hex file. Conveniently, the manual does not specify the details of that hex format or the conversion process, but I'm thinking what you have is right.